rest/authorization-server/src/test/java/org/eclipse/sw360/rest/authserver/GrantTypeClientCredentialsBasicAuthTest.java

```diff
@@ -1,31 +1,65 @@
 /*
- * Copyright Siemens AG, 2019. Part of the SW360 Portal Project.
- *
- * This program and the accompanying materials are made
- * available under the terms of the Eclipse Public License 2.0
- * which is available at https://www.eclipse.org/legal/epl-2.0/
- *
- * SPDX-License-Identifier: EPL-2.0
- */
+* SPDX-FileCopyrightText: © 2024 Siemens AG
+* SPDX-License-Identifier: EPL-2.0
+*/
 package org.eclipse.sw360.rest.authserver;
-
-import org.junit.Before;
+import org.eclipse.sw360.rest.authserver.IntegrationTestBase;
+import org.junit.Test;
 import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.*;
+
+import java.nio.charset.StandardCharsets;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class GrantTypeClientCredentialsBasicAuthTest extends IntegrationTestBase {
+
+    private final String clientId = "trusted-sw360-client";
+    private final String clientSecret = "sw360-secret";
+    private final TestRestTemplate restTemplate = new TestRestTemplate();
+
+    @Test
+    public void successfulTokenRetrieval() {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        headers.setBasicAuth(clientId, clientSecret, StandardCharsets.UTF_8);
+
+        HttpEntity<String> request = new HttpEntity<>("grant_type=client_credentials", headers);
+        String url = "http://localhost:" + port + "/oauth2/token";
+
+        ResponseEntity<String> response = restTemplate.postForEntity(url, request, String.class);
+
+        System.out.println(response.getBody());
```

Comment:
Can we remove the debugging statement from here?

Reply:
Removed